src/Faker/Generator.php

```diff
@@ -193,17 +193,26 @@
  */
 class Generator
 {
-    protected $providers = array();
-    protected $formatters = array();
+    protected static $providers = array();
+    protected static $formatters = array();
```

Comment:
On the other hand, I'm not sure if we should introduce state here.

Personally, in projects where I use `fzaninotto/faker`, I have introduced a [test helper trait](https://github.com/localheinz/test-util/blob/master/src/Helper.php#L34-L47) which handles this. Maybe it would be indeed something that should be solved in user land code, as pointed out by @fzaninotto in https://github.com/fzaninotto/Faker/pull/1005#issuecomment-242667630?